README.md

```diff
@@ -1,148 +1,203 @@
-# Go Test WAF
+# GoTestWAF
 
-GoTestWAF is a tool for API and OWASP attack simulation, that supports a wide range of API protocols including
+GoTestWAF is a tool for API and OWASP attack simulation that supports a wide range of API protocols including
 REST, GraphQL, gRPC, WebSockets, SOAP, XMLRPC, and others.
 
-It was designed to evaluate web application security solutions, such as API security proxies, Web Application Firewalls, IPS, API gateways, and others.
+It was designed to evaluate web application security solutions, such as API security proxies, Web Application Firewalls,
+IPS, API gateways, and others.
 
-# How it works
+## How it works
 
-It is a 3-steps requests generation process that multiply amount of payloads to encoders and placeholders.
-Let's say you defined 2 payloads, 3 encoders (Base64, JSON, and URLencode) and 1 placeholder (HTTP GET variable).
-In this case, the tool will send 2x3x1 = 6 requests in a testcase.
+GoTestWAF generates malicious requests using encoded payloads placed in different parts of HTTP requests: its body, headers,
+URL parameters, etc. Generated requests are sent to the application security tool URL specified during GoTestWAF launch.
```

Comment:
I would suggest to replace "security tool" to "security solution", WAF, "security mechanism".
Security tool is something like nmap, burp, etc.